src/blueapi/service/handler.py

```diff
@@ -80,7 +87,48 @@ def setup_handler(
     config: Optional[ApplicationConfig] = None,
 ) -> None:
     global HANDLER
-    handler = Handler(config)
+
+    provider = None
+    plan_wrappers = []
+
+    if config:
+        visit_service_client: VisitServiceClientBase
+        if config.env.data_writing.visit_service_url is not None:
+            visit_service_client = VisitServiceClient(
+                config.env.data_writing.visit_service_url
+            )
+        else:
+            visit_service_client = LocalVisitServiceClient()
+
+        provider = VisitDirectoryProvider(
+            data_group_name=config.env.data_writing.group_name,
+            data_directory=config.env.data_writing.visit_directory,
+            client=visit_service_client,
+        )
+
+        # Make all dodal devices created by the context use provider if they can
+        try:
+            from dodal.parameters.gda_directory_provider import (
+                set_directory_provider_singleton,
+            )
+
+            set_directory_provider_singleton(provider)
+        except ImportError:
+            logging.error(
+                "Unable to set directory provider for ophyd-async devices, "
+                "a newer version of dodal is required"
```

Comment:
maybe this is too much but, should we specify here instead;

"Unable to set directory provider for ophyd-async devices, "
"most likely due to dodal not being installed from the directory_provider branch"

Reply:
Yeah I wondered about that, depends how soon that changes (i.e. that branch gets merged). I hope relatively soon.

Reply:
ah nevermind then, ignore me